Client/TabManagement/Tab.swift

```diff
@@ -198,7 +198,7 @@ class Tab: NSObject, ThemeApplicable {
         // If the webView doesn't give a title. check the URL to see if it's our Home URL, with no sessionData on this tab.
         // When picking a display title. Tabs with sessionData are pending a restore so show their old title.
         // To prevent flickering of the display title. If a tab is restoring make sure to use its lastTitle.
-        if let url = self.url, InternalURL(url)?.isAboutHomeURL ?? false, sessionData == nil {
```

Comment:
After TabStorage refactor this condition is always false because we restore session data so the Homepage title is returned empty, in the legacy class this is fixed by using `getTabTrayTitle`. I think is safe to fix in displayTitle and avoid having more functions or vars to get the title